lib/kafka/avro_producer.rb

```diff
@@ -25,8 +29,16 @@ def produce(topic, payload, schema_version: 1)
     private
 
     def get_schema(topic, schema_version)
-      schema_path = Rails.root.join('lib', 'kafka', 'schemas', "#{topic}-value-#{schema_version}.avsc")
-      Avro::Schema.parse(File.read(schema_path))
+      if Flipper.enabled?(:kafka_producer_fetch_schema_dynamically)
+        response = @registry.subject_version(topic)
```

Comment:
We need to use the `schema_version` here since the same schema can have multiple versions and it's an argument that the `get_schema` method accepts. It would theoretically default to the latest version but there could be instances we might want to use an earlier version. 

Also, the subject is named after the topic but has a `-value` attached. So something like:
```
@registry.subject_version("#{topic}-value", schema_version)
```

Reply:
good catch. Fixed